src/App.tsx

```diff
@@ -1,26 +1,304 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-const USER_ID = 0;
+type SortSelectOptions = 'all' | 'active' | 'completed';
+export type ErrorMessages =
+  | ''
+  | 'Unable to load todos'
+  | 'Title should not be empty'
+  | 'Unable to add a todo'
+  | 'Unable to delete a todo'
+  | 'Unable to update a todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectSort, setSelectSort] = useState<SortSelectOptions>('all');
+  const [error, setError] = useState<ErrorMessages>('');
+  const [inputTodo, setInputTodo] = useState('');
+  const [loading, setLoading] = useState(false);
+  const inputRefAddTodo = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRefAddTodo.current?.focus();
+  }, [loading]);
```

Comment:
If `loading` is present in the dependency array it should be used!
The `ref` props accepts a callback, maybe this behavior should be moved there:
```js
<Component
  ref={element => {
    if(element) element.focus;
  }}
/>
```

https://react.dev/reference/react-dom/components/common#ref-callback